src/objects/RedditContent.ts

```diff
@@ -0,0 +1,181 @@
+import util from 'util'
+import {defineInspectFunc} from '../helper'
+import {USER_KEYS, SUBREDDIT_KEYS} from '../constants'
+import {isBrowser} from '../helpers'
+import type snoowrap from '../snoowrap'
+import type {Children} from '../interfaces'
+
+/**
+ * A base class for content from reddit. With the expection of Listings, all content types extend this class.
+ * This class should be considered 'abstract', to the extend that JavaScript classes can be -- it should not be necessary to
+ * instantiate it directly.
+ * <style> #RedditContent {display: none} </style>
+ */
+class RedditContent<T extends RedditContent = RedditContent<any>> {
```

Comment:
I'm unsure what you are doing here, making generics for the class in which T extends the Class itself?
and you are not even using that T anywhere in the body

Reply:
It helps constrain the generic when extending other classes.

Reply:
@DedaDev We pass sub classes as generic types to `RedditContent<>` to ensure that methods like `fetch()` and `refresh()` return the correct typing. `RedditContent<any>` is the default in case you do `new RedditContent()`.